Random Users/View Controllers/UsersTableViewController.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  UsersTableViewController.swift
+//  Random Users
+//
+//  Created by Rob Vance on 9/11/20.
+//  Copyright © 2020 Erica Sadun. All rights reserved.
+//
+
+import UIKit
+
+class UsersTableViewController: UITableViewController {
+
+    //MARK: - Properties -
+    let randomUserApiCon = RandomUsersApiController()
+    private let imageFetchQueue = OperationQueue()
+    private let cache = Cache<String, Data>()
+    private var operations = [String : Operation]()
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.randomUserApiCon.fetchUsers { _ in
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    private func loadImage(forCell cell: UsersTableViewCell, forItemAt indexPath: IndexPath) {
+        let user = randomUserApiCon.users[indexPath.row]
+        
+        if let cacheData = cache.value(for: user.email),
+            let image = UIImage(data: cacheData) {
+            cell.imageView?.image = image
+            return
+        }
+        let fetchOperation = FetchImageOperation(user: user)
+        
+        
+        let cacheOperation = BlockOperation {
+            if let data = fetchOperation.imageData {
+                self.cache.cache(key: user.email, value: data)
+            }
+        }
+        
+        let completionOperation = BlockOperation {
+            //switch id to email or phone numbers same as funcs above
+            defer { self.operations.removeValue(forKey: user.email) }
+            if let currentIndexPath = self.tableView.indexPath(for: cell),
+                currentIndexPath != indexPath {
+                print("Got Image")
+                return
+            }
+            if let data = fetchOperation.imageData {
+                cell.imageView?.image = UIImage(data: data)
```

Comment:
I know we discussed this, just make sure to use your actual image view outlet here.